src/combinatorics/partition.lean

```diff
@@ -0,0 +1,172 @@
+/-
```

Comment:
I think this is big enough to become its own PR.

Reply:
I made it in: #4303